src/globus_sdk/_sphinxext.py

```diff
@@ -73,30 +108,41 @@ def gen_rst(self):
 
         yield ""
 
-    def run(self):
-        # implementation is based on the docs for parsing directive content as ReST doc:
-        #   https://www.sphinx-doc.org/en/master/extdev/markupapi.html#parsing-directive-content-as-rest
-        # which is how we will create rst here and have sphinx parse it
 
-        # we need to add content lines into a ViewList which represents the content
-        # of this directive, each one is appended with a "source" name
-        # the best documentation for docutils is the source, so for ViewList, see...
-        #   https://sourceforge.net/p/docutils/code/HEAD/tree/trunk/docutils/docutils/statemachine.py
-        # writing the source name with angle-brackets seems to be the norm, but it's not
-        # clear why -- perhaps docutils will care about the source in some way
-        viewlist = ViewList()
-        for line in self.gen_rst():
-            viewlist.append(line, "<automethodlist>")
+class ListKnownScopes(AddContentDirective):
+    has_content = False
+    required_arguments = 1
+    optional_arguments = 2
 
-        # create a section node (it doesn't really matter what node type we use here)
-        # and add the viewlist we just created as the children of the new node via the
-        # nested_parse method
-        # then return the children (i.e. discard the docutils node)
-        node = nodes.section()
-        node.document = self.state.document
-        nested_parse_with_titles(self.state, viewlist, node)
-        return node.children
+    def gen_rst(self):
+        sb_name = self.arguments[0]
+        sb_basename = sb_name.split(".")[-1]
+
+        add_scopes = []
+        example_scope = None
+        for arg in self.arguments[1:]:
+            if arg.startswith("add_scopes="):
+                add_scopes = [x for x in arg.split("=")[1].split(",")]
+            if arg.startswith("example_scope="):
+                example_scope = arg.split("=")[1]
+        known_scopes = _extract_known_scopes(sb_name) + add_scopes
+        if example_scope is None:
+            example_scope = known_scopes[0]
+
+        yield ""
```

Comment:
I don't necessarily find one better than the other, but one could `yield from [...]` here to be able to write a bunch of lines with just the content and no `yield` prefix.

Reply:
I'm actually somewhat tempted, now that you point this out, to write a helper to `yield from textwrap.dedent(...).split("\n")` on a block of text. And then I could write such things in a single multiline format string, as
```python
yield from _tolines("""...""")
yield from [f"* ``{s}``" for s in known_scopes]
yield from ["", ""]
```

Why don't we save that for now / leave it alone, and I can put in a follow up PR to try that out, to see if it makes the extension code much cleaner?

Reply:
By way of demonstration, I tried this out.
<details><summary>tolines.py example</summary>

```
$ cat tolines.py && echo "---" && python tolines.py
import textwrap


def _tolines(*args):
    for x in args:
        if isinstance(x, str):
            yield from textwrap.dedent(x).split("\n")
        else:
            yield from x


print(
    list(
        _tolines(
            """
            A lovely string
            to report
            about a wonderful thing
            """,
            [f"x={x}" for x in ("a", "b", "c")],
            "",
            "",
        )
    )
)
---
['', 'A lovely string', 'to report', 'about a wonderful thing', '', 'x=a', 'x=b', 'x=c', '', '']
```
</details>